src/main/java/com/minecolonies/coremod/generation/defaults/DefaultEnchanterCraftingProvider.java

```diff
@@ -0,0 +1,403 @@
+import com.minecolonies.coremod.generation.LootTableBuilder;
+import com.minecolonies.coremod.generation.LootTableJsonProvider;
+import net.minecraft.data.DataGenerator;
+import net.minecraft.data.DirectoryCache;
+import net.minecraft.data.IDataProvider;
+import net.minecraft.data.IFinishedRecipe;
+import net.minecraft.enchantment.Enchantment;
+import net.minecraft.enchantment.EnchantmentData;
+import net.minecraft.enchantment.Enchantments;
+import net.minecraft.item.EnchantedBookItem;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.minecolonies.api.util.constant.Constants.MOD_ID;
+
+public class DefaultEnchanterCraftingProvider implements IDataProvider
+{
+    private static final int MAX_BUILDING_LEVEL = 5;
+
+    private final EnchanterRecipeProvider recipeProvider;
+    private final EnchanterLootTableProvider lootTableProvider;
+    private final List<LootTableJson> levels;
+
+    public DefaultEnchanterCraftingProvider(@NotNull final DataGenerator generatorIn)
+    {
+        levels = new ArrayList<>();
+
+        // building level 1
+        levels.add(new LootTableBuilder()
+                .item(enchantedBook(Enchantments.AQUA_AFFINITY, 1), 50)
+                .item(enchantedBook(Enchantments.BANE_OF_ARTHROPODS, 1), 50)
+                .item(enchantedBook(Enchantments.BLAST_PROTECTION, 1), 50)
+                .item(enchantedBook(Enchantments.DEPTH_STRIDER, 1), 50)
+                .item(enchantedBook(Enchantments.FEATHER_FALLING, 1), 50)
+                .item(enchantedBook(Enchantments.FIRE_ASPECT, 1), 50)
+                .item(enchantedBook(Enchantments.FIRE_PROTECTION, 1), 50)
+                .item(enchantedBook(Enchantments.FLAME, 1), 50)
+                .item(enchantedBook(Enchantments.FROST_WALKER, 1), 50)
+                .item(enchantedBook(Enchantments.KNOCKBACK, 1), 50)
+                .item(enchantedBook(Enchantments.LOOTING, 1), 50)
+                .item(enchantedBook(Enchantments.POWER, 1), 50)
+                .item(enchantedBook(Enchantments.PROJECTILE_PROTECTION, 1), 50)
+                .item(enchantedBook(Enchantments.PROTECTION, 1), 50)
+                .item(enchantedBook(Enchantments.PUNCH, 1), 50)
+                .item(enchantedBook(Enchantments.RESPIRATION, 1), 50)
+                .item(enchantedBook(Enchantments.SHARPNESS, 1), 50)
+                .item(enchantedBook(Enchantments.SMITE, 1), 50)
+                .item(enchantedBook(Enchantments.SWEEPING, 1), 50)
+                .item(enchantedBook(Enchantments.UNBREAKING, 1), 50)

```

Comment:
These are copied from the previous configuration, but there are some issues that we might want to address:

1. They're all given equal probability.  The enchantment configuration defines a "rarity" that in turn determines a weighting value that we should perhaps be applying to influence this instead.
2. Some vanilla enchants are entirely missing from this list:
    * Soul Speed
    * Efficiency
    * Silk Touch
    * Fortune
    * Luck of the Sea
    * Lure
    * Loyalty
    * Impaling
    * Riptide
    * Channeling
    * Multishot
    * Quick Charge
    * Piercing
3. Which also highlights an unfortunate side effect of doing things this way (rather than looking up enchants at runtime): no support for modded enchants (without datapack) and even vanilla enchants have to be added manually before they're supported.  Though the advantage is that we can more deliberately filter and alter probabilities of specific enchants this way.

Reply:
Efficiency should be added with the same probabilities and levels as sharpness.
Multishot can be as rare as mending
Silk Touch the same as mending
Fortune can be as rare as mending, but with increasing levels ( 1 at level 3, 2 at level 4, and 3 at level 5)
Quick Charge could be the same as sweeping

The rest we can leave out for now.